packages/frontend/src/pages/SavedExplorer.tsx

```diff
@@ -1,96 +1,120 @@
-import { Card } from '@blueprintjs/core';
-import React, { useEffect } from 'react';
+import { Card, NonIdealState, Spinner } from '@blueprintjs/core';
+import React from 'react';
 import { useHistory, useParams } from 'react-router-dom';
 import AboutFooter from '../components/AboutFooter';
 import { Explorer } from '../components/Explorer';
 import ExplorePanel from '../components/Explorer/ExplorePanel/index';
 import { useSavedQuery } from '../hooks/useSavedQuery';
-import { useExplorer } from '../providers/ExplorerProvider';
+import { ExplorerProvider } from '../providers/ExplorerProvider';
 
 const SavedExplorer = () => {
     const history = useHistory();
-    const pathParams = useParams<{ savedQueryUuid: string }>();
-    const {
-        actions: { setState, reset },
-    } = useExplorer();
-    const { data } = useSavedQuery({ id: pathParams.savedQueryUuid });
+    const pathParams =
+        useParams<{ savedQueryUuid: string; projectUuid: string }>();
+    const { data, isLoading, error } = useSavedQuery({
+        id: pathParams.savedQueryUuid,
+    });
     const onBack = () => {
-        reset();
         history.push({
-            pathname: `/saved`,
+            pathname: `/projects/${pathParams.projectUuid}/home`,
         });
     };
 
-    useEffect(() => {
-        if (data) {
-            setState({
-                shouldFetchResults: true,
-                chartName: data.name,
-                tableName: data.tableName,
-                dimensions: data.metricQuery.dimensions,
-                metrics: data.metricQuery.metrics,
-                filters: data.metricQuery.filters,
-                sorts: data.metricQuery.sorts,
-                limit: data.metricQuery.limit,
-                columnOrder: data.tableConfig.columnOrder,
-                selectedTableCalculations:
-                    data.metricQuery.tableCalculations.map((t) => t.name),
-                tableCalculations: data.metricQuery.tableCalculations,
-                additionalMetrics: data.metricQuery.additionalMetrics,
-            });
-        }
-    }, [data, setState]);
+    if (isLoading) {
+        return (
+            <div style={{ marginTop: '20px' }}>
+                <NonIdealState title="Loading..." icon={<Spinner />} />
+            </div>
+        );
+    }
+    if (error) {
+        return (
+            <div style={{ marginTop: '20px' }}>
+                <NonIdealState
+                    title="Unexpected error"
+                    description={error.error.message}
+                />
+            </div>
+        );
+    }
 
     return (
-        <div
-            style={{
-                display: 'flex',
-                flexDirection: 'row',
-                flexWrap: 'nowrap',
-                justifyContent: 'stretch',
-                alignItems: 'flex-start',
-            }}
+        <ExplorerProvider
+            initialState={
+                data
+                    ? {
```

Comment:
the initialSate looks similar to data, is it not more legible to use the same model here ?

Reply:
yes, I think that the provider state could be an extension of the saved chart model. I'm not doing that change in this PR since it would be quite a big refactor. Adding this to my TODO list.